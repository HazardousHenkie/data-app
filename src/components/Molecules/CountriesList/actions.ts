import { action } from 'typesafe-actions'

import ActionTypes from './constants'

export const getCountriesData = () => action(ActionTypes.GET_COUNTRIES_DATA)
export const getCountriesDataSuccess = (countriesData: []) =>
    action(ActionTypes.GET_COUNTRIES_DATA_SUCCESS, countriesData)
export const getCountriesDataError = (error: Error) =>
    action(ActionTypes.GET_COUNTRIES_DATA_ERROR, error)
