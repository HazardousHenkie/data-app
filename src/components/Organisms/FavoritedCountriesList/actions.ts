import { action } from 'typesafe-actions'

import ActionTypes from './constants'

import { CountryState } from './types'

const setSelectedCountry = (country: CountryState['country']) =>
    action(ActionTypes.GET_FAVORITED_COUNTRIES, country)

export default setSelectedCountry
