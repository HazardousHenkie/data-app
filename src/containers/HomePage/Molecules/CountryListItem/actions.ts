import { action } from 'typesafe-actions'

import ActionTypes from './constants'

import CountryState from './types'

const setSelectedCountry = (country: CountryState['country']) =>
    action(ActionTypes.SET_SELECTED_COUNTRY, country)

export default setSelectedCountry
