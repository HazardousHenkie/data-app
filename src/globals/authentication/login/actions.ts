import { action } from 'typesafe-actions'

import AuthenticationState from '../types'
import ActionTypes from './constants'

export const loginRequest = (loginToken: string) =>
    action(ActionTypes.LOGIN_REQUEST, loginToken)
export const loginSuccess = (user: AuthenticationState['user']) =>
    action(ActionTypes.LOGIN_SUCCESS, user)
export const loginError = (error: Error | boolean) =>
    action(ActionTypes.LOGIN_ERROR, error)
