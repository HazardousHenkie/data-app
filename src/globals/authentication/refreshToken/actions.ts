import { action } from 'typesafe-actions'

import AuthenticationState from '../types'
import ActionTypes from './constants'

export const getRefreshTokenRequest = (id: string) =>
    action(ActionTypes.GET_REFRESH_TOKEN_REQUEST, id)
export const getRefreshTokenSuccess = (user: AuthenticationState['user']) =>
    action(ActionTypes.GET_REFRESH_TOKEN_SUCCESS, user)
export const getRefreshTokenError = (error: Error | boolean) =>
    action(ActionTypes.GET_REFRESH_TOKEN_ERROR, error)
