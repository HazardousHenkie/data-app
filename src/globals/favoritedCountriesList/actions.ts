import { action } from 'typesafe-actions'

import { ResponseError } from 'utils/request'
import ActionTypes from './constants'

import FavoritedCountriesState from './types'

export const getFavoritedCountries = () =>
    action(ActionTypes.GET_FAVORITED_COUNTRIES)

export const setFavoritedCountries = (
    favoritedCountries: FavoritedCountriesState['countries']
) => action(ActionTypes.SET_FAVORITED_COUNTRIES, favoritedCountries)

export const getFavoritedCountriesError = (error: ResponseError) =>
    action(ActionTypes.GET_FAVORITED_COUNTRIES_ERROR, error)
