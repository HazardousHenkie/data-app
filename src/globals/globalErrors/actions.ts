import { action } from 'typesafe-actions'

import ActionTypes from './constants'

export const getErrors = () => action(ActionTypes.GET_ERRORS)

export const setError = (error: Error) => action(ActionTypes.SET_ERROR, error)

export const removeError = () => action(ActionTypes.REMOVE_ERROR)

export const setErrors = (errors: Error[]) =>
    action(ActionTypes.SET_ERRORS, errors)
