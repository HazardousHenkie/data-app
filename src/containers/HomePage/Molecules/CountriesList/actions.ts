import { action } from 'typesafe-actions'

import { CountryInterface } from 'containers/HomePage/Molecules/CountryListItem/types'
import ActionTypes from './constants'

export const getCountriesData = () => action(ActionTypes.GET_COUNTRIES_DATA)
export const getCountriesDataSuccess = (countriesData: CountryInterface[]) =>
    action(ActionTypes.GET_COUNTRIES_DATA_SUCCESS, countriesData)
export const getCountriesDataError = (error: Error) =>
    action(ActionTypes.GET_COUNTRIES_DATA_ERROR, error)
