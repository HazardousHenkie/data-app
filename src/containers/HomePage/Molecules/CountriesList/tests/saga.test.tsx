import { put, takeLatest } from 'redux-saga/effects'

import Actions from '../constants'
import countriesListRootSaga, { getCountriesDataSaga } from '../saga'
import { getCountriesDataSuccess, getCountriesDataError } from '../actions'

describe('getCountriesData Saga', () => {
    let getCountriesDataGenerator: Generator

    beforeEach(() => {
        getCountriesDataGenerator = getCountriesDataSaga()

        const callDescriptor = getCountriesDataGenerator.next().value
        expect(callDescriptor).toMatchSnapshot()
    })

    it('should dispatch the getCountriesData action if it requests the data successfully', () => {
        const response = [
            {
                alpha2Code: 'AF',
                alpha3Code: 'AFG',
                altSpellings: ['AF', 'Afġānistān']
            }
        ]
        const putDescriptor = getCountriesDataGenerator.next(response).value

        // eslint-disable-next-line redux-saga/no-unhandled-errors
        expect(putDescriptor).toEqual(put(getCountriesDataSuccess(response)))
    })

    it('should call the repoLoadingError action if the response errors', () => {
        const response = new Error('Some error')
        const putDescriptor = getCountriesDataGenerator.throw(response).value

        // eslint-disable-next-line redux-saga/no-unhandled-errors
        expect(putDescriptor).toEqual(put(getCountriesDataError(response)))
    })
})

describe('signInSaga Saga', () => {
    const countriesList = countriesListRootSaga()

    it('should start task to watch for GET_COUNTRIES_DATA action.', () => {
        const takeLatestDescriptor = countriesList.next().value

        expect(takeLatestDescriptor).toEqual(
            takeLatest(Actions.GET_COUNTRIES_DATA, getCountriesDataSaga)
        )
    })
})
