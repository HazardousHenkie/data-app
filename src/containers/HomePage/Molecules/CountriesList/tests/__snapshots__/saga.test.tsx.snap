// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCountriesData Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://restcountries.eu/rest/v2/all",
      Object {
        "method": "GET",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getCountriesData Saga should dispatch the getCountriesData action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "https://restcountries.eu/rest/v2/all",
      Object {
        "method": "GET",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
